python/ray/data/dataset.py

```diff
@@ -338,6 +341,10 @@ def parse_filename(row: Dict[str, Any]) -> Dict[str, Any]:
             num_gpus: The number of GPUs to reserve for each parallel map worker. For
                 example, specify `num_gpus=1` to request 1 GPU for each parallel map
                 worker.
+            concurrency: The number of Ray actors to use. For a fixed-sized actors pool
+                of size ``n``, specify ``concurrency=n``. For a autoscaling actors pool
+                from ``m`` to ``n`` actors, specify ``concurrency=(m, n)``.
+                ``None`` by default to use Ray tasks.
```

Comment:
I think we should decide whether to use tasks or actors based on the UDF, i.e., use actors for stateful UDFs (classes), tasks for stateless (functions). 
Because for task-based operators, we'll also want to control the concurrency. 
For the documentation, we can say "concurrency" controls the number of concurrent workers.

Reply:
+1. I think this is also less confusing, since function->task, class->actor is pretty natural in the rest of Ray.

Reply:
Yes, I agree. Changed the code to decide task/actor based on UDF. I didn't change in the first place, because we allow run function with actor. With this PR, this behavior will be disallowed, and throw an error.